File: src/test/java/no/nav/bidrag/beregn/bidragsevne/BidragsevneberegningGrunnlagTest.java
package no.nav.bidrag.beregn.bidragsevne;

import static org.assertj.core.api.Assertions.assertThat;

import java.math.BigDecimal;
import java.util.List;
import no.nav.bidrag.beregn.TestUtil;
import no.nav.bidrag.beregn.felles.SjablonUtil;
import no.nav.bidrag.beregn.felles.bo.Sjablon;
import no.nav.bidrag.beregn.felles.enums.SjablonNavn;
import no.nav.bidrag.beregn.felles.enums.SjablonTallNavn;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

@DisplayName("Test hent av sjablonverdi")
class BidragsevneberegningGrunnlagTest {

  private final List<Sjablon> sjablonListe = TestUtil.byggSjabloner();

  @Test
  void hentSjablon() {

    var sjablonVerdi =
        SjablonUtil.hentSjablonverdi(sjablonListe, SjablonTallNavn.SKATTESATS_ALMINNELIG_INNTEKT_PROSENT);
    assertThat(sjablonVerdi).isEqualTo(BigDecimal.valueOf(22));

    var sortertTrinnvisSkattesatsListe = SjablonUtil
        .hentTrinnvisSkattesats(sjablonListe, SjablonNavn.TRINNVIS_SKATTESATS);

    assertThat(sortertTrinnvisSkattesatsListe.size()).isEqualTo(4);
    assertThat(sortertTrinnvisSkattesatsListe.get(0).getInntektGrense()).isEqualTo(BigDecimal.valueOf(180800));
    assertThat(sortertTrinnvisSkattesatsListe.get(0).getSats()).isEqualTo(BigDecimal.valueOf(1.9));

  }
}

File: src/test/kotlin/no/nav/bidrag/beregn/bidragsevne/BidragsevneBeregningGrunnlagTest.kt
package no.nav.bidrag.beregn.bidragsevne

import no.nav.bidrag.beregn.TestUtil.byggSjabloner
import no.nav.bidrag.beregn.felles.util.SjablonUtil.hentSjablonverdi
import no.nav.bidrag.beregn.felles.util.SjablonUtil.hentTrinnvisSkattesats
import no.nav.bidrag.domene.enums.sjablon.SjablonNavn
import no.nav.bidrag.domene.enums.sjablon.SjablonTallNavn
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions.assertAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import java.math.BigDecimal

internal class BidragsevneBeregningGrunnlagTest {
    private val sjablonListe = byggSjabloner()

    @Test
    fun hentSjablon() {
        val sjablonverdiInntekt =
            hentSjablonverdi(
                sjablonListe = sjablonListe,
                sjablonTallNavn = SjablonTallNavn.SKATTESATS_ALMINNELIG_INNTEKT_PROSENT,
            )
        val sortertTrinnvisSkattesatsListe =
            hentTrinnvisSkattesats(sjablonListe = sjablonListe, sjablonNavn = SjablonNavn.TRINNVIS_SKATTESATS)

        assertAll(
            Executable { assertThat(sjablonverdiInntekt).isEqualTo(BigDecimal.valueOf(22)) },
            Executable { assertThat(sortertTrinnvisSkattesatsListe).hasSize(4) },
            Executable { assertThat(sortertTrinnvisSkattesatsListe[0].inntektGrense).isEqualTo(BigDecimal.valueOf(180800)) },
            Executable { assertThat(sortertTrinnvisSkattesatsListe[0].sats).isEqualTo(BigDecimal.valueOf(1.9)) },
        )
    }
}
